scripts/get-terraform-supported-versions.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+get_last_day_of_month() {
+	last_day_of_month=0
+	case $1 in
+		01|03|05|07|08|10|12)
+			last_day_of_month=31 ;;
+		04|06|09|11)
+			last_day_of_month=30 ;;
+		02)
+			# February: check if it's a leap year
+			if (( year % 4 == 0 && ( year % 100 != 0 || year % 400 == 0 ) )); then
+				last_day_of_month=29
+			else
+				last_day_of_month=28
+			fi
+			;;
+		esac
+	echo $last_day_of_month
+}
+
+
+add_end_support_date() {
+	new_json_list=$(echo "$1" | jq -c '.[]' | while IFS= read -r obj; do
+		input_date=$(echo "$obj" | jq -r '.published_at')
+
+		# Extract the year, month, day, hour, minute, and second from the input date
+		year="${input_date:0:4}"
+		month="${input_date:5:2}"
+		hour="${input_date:11:2}"
+		minute="${input_date:14:2}"
+		second="${input_date:17:2}"
+		last_day_of_month=$(get_last_day_of_month "$month")
+		new_year=$((year + 2))
+
+		new_date="${new_year}-${month}-${last_day_of_month}T${hour}:${minute}:${second}Z"
+
+		echo "$obj" | jq --arg new_date "$new_date" '.end_support_date = $new_date'
+	done | jq -s '.')
+
+	echo "$new_json_list"
+}
+
+page=1
+api_version="2022-11-28"
+
+while true; do
```

Comment:
did you run shellcheck on this script?

Reply:
Yes I did run it. It's also run in the CI

Reply:
Thanks!